spark/src/main/scala/org/apache/spark/sql/delta/files/TahoeFileIndex.scala

```diff
@@ -126,6 +127,19 @@ abstract class TahoeFileIndex(
     addFile.defaultRowCommitVersion.foreach(defaultRowCommitVersion =>
       metadata.put(DefaultRowCommitVersion.METADATA_STRUCT_FIELD_NAME, defaultRowCommitVersion))
 
+    // Attach filter types to FileActions, so that later [[DeltaParquetFileFormat]] could pick it up
+    // to decide which kind of rows should be filtered out. This info is necessary for reading CDC
+    // rows that have been deleted (marked in DV), in which case marked rows must be kept rather
+    // than filtered out. In such a case, the `filterTypes` map will be populated by [[CDCReader]]
+    // to indicate IF_NOT_CONTAINED filter should be used. In other cases, `filterTypes` will be
+    // empty, so we generate IF_CONTAINED as the default DV behavior
```

Comment:
Move this comment just before `val filterType`

Also it could be a bit shorter, e.g.:
> Set the filter type to `IF_CONTAINED` by default to filter out rows unless a filter type was explicitly provided in `rowIndexFilters`. This can happen e.g. when reading CDC data to keep deleted rows instead of filtering them out.

Reply:
Not addressed ^

Reply:
@johanl-db I actually addressed, but I was not sure if your version removes too much original info, @xupefei, is it fine if we keep it like

> Set the filter type to IF_CONTAINED by default to filter out rows unless a filter type was explicitly provided in rowIndexFilters. This can happen e.g. when reading CDC data to keep deleted rows instead of filtering them out.

Reply:
I would suggest
> Set the filter type to IF_CONTAINED by default to let [[DeltaParquetFileFormat]] filter out rows unless a filter type was explicitly provided in rowIndexFilters. This can happen e.g. when reading CDC data to keep deleted rows instead of filtering them out.